### PR TITLE
[prometheus-smartctl-exporter] Initial stab at a prometheus-smartctl-exporter chart

### LINKED ISSUE #2356: smartctl_exporter Chart
It would be great to have an official helm chart for this project.

I have a rough one started here:
https://github.com/pnnl-miscscripts/miscscripts/tree/master/charts/charts/smartctl-exporter
That could be donated.

It has a bit of tricky logic in it to support having multiple daemonsets configured differently in case you have different types of nodes in the cluster.

### PR DESCRIPTION
**Pull request by kfox1111**
#### Which issue this PR fixes

- fixes #2356

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

File: charts/prometheus-smartctl-exporter/Chart.yaml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+# This is the chart version. This version number should be incremented each time you make changes
+# to the chart and its templates, including the app version.
+# Versions are expected to follow Semantic Versioning (https://semver.org/)
+version: 0.1.0
+
+# This is the version number of the application being deployed. This version number should be
+# incremented each time you make changes to the application. Versions are not expected to
+# follow Semantic Versioning. They should reflect the version the application is using.
+# It is recommended to use it with quotes.
+appVersion: "0.7"
```
Comment (monotek) [inline charts/prometheus-smartctl-exporter/Chart.yaml:24]:
```suggestion
appVersion: "v0.7.0"
```

**Review by monotek — CHANGES_REQUESTED** [review]
Please fix linting errors, add defaults to notes.txt and change the appversion in chart.yaml

File: charts/prometheus-smartctl-exporter/values.yaml [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+  # requests:
+  #  cpu: 100m
+  #  memory: 128Mi
+
+nodeSelector: {}
+
+tolerations:
+- key: CriticalAddonsOnly
+  operator: Exists
+- operator: Exists
```
Comment (monotek) [inline charts/prometheus-smartctl-exporter/values.yaml:71]: This looks not right to me. Key, operator and effect should be in one entry.

**Review by monotek — CHANGES_REQUESTED** [review]

File: charts/prometheus-smartctl-exporter/values.yaml (hunk 2 above)
Comment (kfox1111) [reply charts/prometheus-smartctl-exporter/values.yaml:71]: It was originally copied from kube-proxy. as of k8s 1.23 looks like they still have that toleration....

I could change it though if you think that's better. Probably need to put in toleration for the control plane explicitly then.

Comment (kfox1111) [reply charts/prometheus-smartctl-exporter/values.yaml:71]: How about we change it to match the node exporter charts values?
https://github.com/prometheus-community/helm-charts/blob/main/charts/prometheus-node-exporter/values.yaml#L160-L162

Comment (kfox1111) [reply charts/prometheus-smartctl-exporter/values.yaml]: I submitted that change.

File: charts/prometheus-smartctl-exporter/values.yaml [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+  #  cpu: 100m
+  #  memory: 128Mi
+  # requests:
+  #  cpu: 100m
+  #  memory: 128Mi
+
+nodeSelector: {}
+
+tolerations:
+- effect: NoSchedule
```
Comment (monotek) [inline charts/prometheus-smartctl-exporter/values.yaml:69]: Can you please fix indentation too (move 2 spaces right)

**Review by monotek — CHANGES_REQUESTED** [review]

**Review by monotek — APPROVED** [review]

**Comment by onedr0p** [comment]
```
docker pull quay.io/prometheus/smartctl-exporter:v0.7.0
Error response from daemon: unauthorized: access to the requested resource is not authorized
```

Hi! The image doesn't appear to be available for this.

**Comment by kfox1111** [comment]
Workaround proposed here: https://github.com/prometheus-community/helm-charts/pull/2365

**Pull request merged**